src/jit/lowerarm64.cpp

```diff
@@ -584,97 +579,107 @@ void Lowering::TreeNodeInfoInit(GenTree* stmt)
+                                // computed in the register in which the argument is passed to the call.
+                                putArgChild ->gtLsraInfo.setSrcCandidates(l, targetMask);
+
+                                // We consume one source for each item in this list
+                                info->srcCount++;
+                                iterationNum++;
+
+                                // Update targetReg and targetMask for the next putarg_reg (if any)
+                                targetReg  = REG_NEXT(targetReg);
+                                targetMask = genRegMask(targetReg);   
+                            }
                         }
                         else
                         {
-                            assert(!"Can't predict unsupported TYP_STRUCT arg kind");
+                            noway_assert(!"Unsupported TYP_STRUCT arg kind");
                         }
 
-                        unsigned slots = ((unsigned)(roundUp(originalSize, TARGET_POINTER_SIZE))) / REGSIZE_BYTES;
-                        regNumber reg = (regNumber)(argReg + 1);
-                        unsigned remainingSlots = slots - 1;
-
-                        if (remainingSlots > 1)
-                        {
-                            NYI_ARM64("Lower - Struct typed arguments (size>16)");
-                        }
+                        unsigned  slots   = ((unsigned)(roundUp(originalSize, REGSIZE_BYTES))) / REGSIZE_BYTES;
+                        regNumber curReg  = argReg;
+                        regNumber lastReg = argIsFloat ? REG_ARG_FP_LAST : REG_ARG_LAST;
+                        unsigned remainingSlots = slots;
 
-                        while (remainingSlots > 0 && reg <= REG_ARG_LAST)
+                        while (remainingSlots > 0)
                         {
-                            argMask |= genRegMask(reg);
-                            reg = (regNumber)(reg + 1);
+                            argMask |= genRegMask(curReg);
                             remainingSlots--;
-                            regCount++;
-                        }
 
-                        if (remainingSlots > 1)
-                        {
-                            NYI_ARM64("Lower - Struct typed arguments (Reg/Stk split)");
-                        }
+                            if (curReg == lastReg)
+                                break;
 
-                        short internalIntCount = 0;
-                        if (remainingSlots > 0)
-                        {
-                            // This TYP_STRUCT argument is also passed in the outgoing argument area
-                            // We need a register to address the TYP_STRUCT
-                            // And we may need 2
-                            internalIntCount = 2;
+                            curReg = REG_NEXT(curReg);
                         }
-                        argNode->gtLsraInfo.internalIntCount = internalIntCount;
+
```

Comment:
It appears that this second loop is a bit redundant, and could be combined with the above - is it not the case that, after this `while (remainingSlots > 0)` loop, we have simply computed `argMask` and it should be equal to `targetMask` that we computed above? If that's the case, then perhaps the second loop could be done only under DEBUG - if not, then perhaps some additional comments are needed.